static/js/components/tiles/sv_ranking_units.tsx

```diff
@@ -163,71 +139,133 @@ export function SvRankingUnits(props: SvRankingUnitsProps): JSX.Element {
 
   return (
     <React.Fragment>
-      {showHighest && (
+      {rankingMetadata.showHighest && (
         <div className="ranking-unit-container highest-ranking-container">
           {props.isDataTile && (
             <div
               className={DATA_CSS_CLASS}
-              data-csv={rankingPointsToCsv(points, svNames || [svName])}
+              data-csv={rankingPointsToCsv(
+                rankingGroup.points,
+                rankingGroup.svName
+              )}
             />
           )}
-          <RankingUnit
-            key={`${statVar}-highest`}
-            unit={unit}
-            forwardRef={highestRankingUnitRef}
-            scaling={scaling}
-            title={formatString(
-              title || (highestTitle ? highestTitle : "Highest ${statVar}"),
-              {
-                date: dateRange,
-                placeName: "",
-                statVar: svName,
-              }
-            )}
-            points={points.slice(-rankingCount).reverse()}
-            isHighest={true}
-            svNames={isMultiColumn ? svNames : undefined}
-            formatNumberFn={formatNumber}
-          />
+          {getRankingUnit(
+            title,
+            statVar,
+            rankingGroup,
+            rankingMetadata,
+            true,
+            highestRankingUnitRef
+          )}
           <ChartFooter
-            sources={sources}
-            handleEmbed={() => handleEmbed(points.reverse(), sources, true)}
+            sources={rankingGroup.sources}
+            handleEmbed={() => handleEmbed(true)}
           />
         </div>
       )}
-      {showLowest && (
+      {rankingMetadata.showLowest && (
         <div className="ranking-unit-container lowest-ranking-container">
           {props.isDataTile && (
             <div
               className={DATA_CSS_CLASS}
-              data-csv={rankingPointsToCsv(points, svNames || [svName])}
+              data-csv={rankingPointsToCsv(
+                rankingGroup.points,
+                rankingGroup.svName
+              )}
             />
           )}
-          <RankingUnit
-            key={`${statVar}-lowest`}
-            unit={unit}
-            forwardRef={lowestRankingUnitRef}
-            scaling={scaling}
-            title={formatString(
-              title || (lowestTitle ? lowestTitle : "Lowest ${statVar}"),
-              {
-                date: dateRange,
-                placeName: "",
-                statVar: svName,
-              }
-            )}
-            numDataPoints={numDataPoints}
-            points={points.slice(0, rankingCount)}
-            isHighest={false}
-            svNames={isMultiColumn ? svNames : undefined}
-            formatNumberFn={formatNumber}
-          />
+          {getRankingUnit(
```

Comment:
What do you think about making `getRankingUnit` a proper component? Maybe `<StatVarRankingUnit {...props} />`

Reply:
`getRankingUnit` does very few calculations and just renders a `RankingUnit` component so not sure it warrants its own component. I extracted it out as a function where most arguments come from props or datafetch so that it can be easily used by server to ensure any changes to the ranking tile will also result in a change in server. Let me know if you feel strongly about changing it into a component.

Reply:
The function works for me